test/integration/short/client-pool-tests.js

```diff
@@ -707,6 +697,47 @@ describe('Client', function () {
         }
       ], done);
     });
+    it('should reconnect in the background', function (done) {
+      var client = newInstance({
+        pooling: { heartBeatInterval: 0, warmup: true },
+        policies: { reconnection: new policies.reconnection.ConstantReconnectionPolicy(1000) }
+      });
+      async.series([
+        client.connect.bind(client),
+        function (next) {
+          var hosts = client.hosts.values();
+          assert.strictEqual('1', helper.lastOctetOf(client.controlConnection.host));
+          assert.strictEqual(3, hosts.length);
+          hosts.forEach(function (h) {
+            assert.ok(h.pool.connections.length > 0, 'with warmup = true, it should create the core connections');
+          });
+          next();
+        },
+        helper.toTask(helper.ccmHelper.stopNode, null, 1),
+        helper.toTask(helper.ccmHelper.stopNode, null, 2),
+        helper.toTask(helper.ccmHelper.stopNode, null, 3),
+        function tryToQuery(next) {
+          async.timesSeries(10, function (n, timesNext) {
+            client.execute(helper.queries.basic, function (err) {
+              //it should fail
+              assert.ok(err);
+              timesNext();
+            });
+          }, next);
+        },
+        function startNode3(next) {
+          helper.ccmHelper.startNode(3, helper.wait(5000, next));
```

Comment:
Unfortunately 5 seconds doesn't seem to be enough time for our jenkins environment in some cases.  I've also seen this test fail because nodes remove not stopped and removed in previous tests.  I'll see if there's something we can do like listen for 'on' events for a host and then timeout after a prohibitively long time (like 30 seconds?).